### ISSUE TITLE
Scheduler not restarting

### ISSUE THREAD

**Issue by ppanero** [MISP/MISP#2075]
### Work environment

| Questions                 | Answers
|---------------------------|--------------------
| Type of issue             | Bug
| OS version (server)       | CentOS 7
| OS version (client)       | -
| PHP version               | 5.6
| MISP version / git hash   | 2.4.69
| Browser                   | -

### Expected behavior

Worker quitted by itself. Should've been restarted by rc.local or puppet. 

### Actual behavior
Didn't restart

### Steps to reproduce the behavior
kill scheduler process

### Logs, screenshots, configuration dump, ...
Checking in redis:
```
keys *
1) "resque:worker:misp-dev:24272:prio:started"
2) "resque:ResqueWorker"
3) "resque:ResqueSchedulerWorker"
4) "resque:workerLogger"
5) "resque:worker:misp-dev:24254:default:started"
6) "resque:stat:processed"
7) "resque:worker:misp-dev:24308:email:started"
8) "resque:worker:misp-dev:24290:cache:started"
9) "resque:workers"
```
the scheduler stopped being there:
 - Previous to scheduler death/kill
```
redis-cli smembers resque:workers
1) "misp-dev:23959:_schdlr_"
2) "misp-dev:23979:cache"
3) "misp-dev:23961:prio"
4) "misp-dev:23943:default"
5) "misp-dev:23997:email"
```
 - After
```
redis-cli smembers resque:workers
1) "misp-dev:24308:email"
2) "misp-dev:24272:prio"
3) "misp-dev:24254:default"
4) "misp-dev:24290:cache"
```
However the key "resque:ResqueSchedulerWorker" is still present and therefore when running the start.sh script it shows that the scheduler is already running:
```
Stopping workers
Stopping 23979 ... Done
Stopping 23961 ... Done
Stopping 23943 ... Done
Stopping 23997 ... Done

Creating workers
Starting worker ... Done

Creating workers
Starting worker ... Done

Creating workers
Starting worker ... Done

Creating workers
Starting worker ... Done

Creating the scheduler workers
The scheduler worker is already running
```

When it is clearly not:
```
get resque:ResqueSchedulerWorker
"24015"
```
```
ps aux | grep 24015
root     24340  0.0  0.0 112648   960 pts/2    S+   14:00   0:00 grep --color=auto 24015
```

Note that the same behaviour applies to all the workers. However since there is no separate key for the others (i.e. the resque:ResqueSchedulerWorker) this does not happen.

In addition. Dead workers appear in the UI when they have been killed and are no longer in redis. This seems an issue from: https://github.com/MISP/MISP/blob/e1ea9cb300ca17d51a12ee059b7420168f7b4a30/app/Model/Server.php#L2950

**Comment by iglocska** [comment]
Interesting. It is for sure wonky and we already had to go around the suggested best practices to get the current worker output. I wouldn't do anything besides quick fixes to it anymore though, we have a new background processing package in the pipe written completely from the ground-up by @Rafiot that we want to replace cakeresque with.

**Comment by ppanero** [comment]
@iglocska Oh cool. Then Ill just quick fix it on the start.sh maybe kinda deleting the scheduler key and restarting or something; and wait for it to come out :D 

thanks!

**Comment by iglocska** [comment]
Yeah I think that would be best :) Alternatively we can add a redis DEL to
manually purge the scheduler on restart.

**Comment by ppanero** [comment]
@iglocska  yeah that's what i did :D thanks

**Comment by adulau** [comment]
@ppanero our replacement tool (still in dev) is available there https://github.com/MISP/SimpleQueue. Feedback welcome ;-)

**Comment by ppanero** [comment]
@adulau cool! Will check it out, and if you need a hand on a specific thing just tell me :D will gladly contribute! We can talk about it on gitter :D

**Comment by Rafiot** [comment]
Glad you're interested! 
We need to discuss that with @iglocska too, this will be a massive change and will not happen before the objects.

**Comment by icepaule** [comment]
Hi there, just from having the same issue:

_service misp-workers stop
sudo -u www-data ../cake CakeResque.CakeResque clear
sudo -u www-data ../cake CakeResque.CakeResque reset
service misp-workers start_

... helped at my end of the wire.  ;-)
